modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -456,6 +456,21 @@ public void visit(BallerinaFunction function) {
 
             BlockStmt blockStmt = function.getCallableUnitBody();
             blockStmt.accept(this);
+
+            if (function.getReturnParameters().length > 0) {
+                Statement[] stmts = blockStmt.getStatements();
+                for (int i = 0; i < stmts.length; i++) {
+                    if (stmts[i].resultsInAReturn()) {
+                        checkUnreachableStmt(stmts, i + 1);
+                    }
+                }
+
+                if (!blockStmt.resultsInAReturn()) {
+                    throw new SemanticException(function.getNodeLocation().getFileName() + ":" +
```

Comment:
Can we put these error messages to the messages.properties file and throw the error as illustrated below?

BLangExceptionHelper.throwSemanticError(connectorDef,
                        SemanticErrors.REDECLARED_SYMBOL, connectorName);